### PR TITLE
Update dynamic fields responders after change

### PR DESCRIPTION
**Pull request by romanblanco**
https://bugzilla.redhat.com/show_bug.cgi?id=1614436

Removing field changed from dynamic to static from the list of responders for each associated field.

Steps to reproduce: 

screencast from 2018-09-13 23-55-38

### CODE REVIEW

**Comment by romanblanco** [comment]
@miq-bot add_label bug, blocker, gaprindashvili/yes

**Comment by d-m-u** [comment]
Thanks, @romanblanco !! 🎉

File: src/dialog-editor/components/modal-field/field.html [hunk 1]
```diff
@@ -42,6 +42,7 @@ <h4 class="modal-title" id="myModalLabel" translate>Edit Field Details</h4>
            pf-label="{{'Dynamic'|translate}}">
         <input bs-switch
                ng-model="vm.modalData.dynamic"
+               ng-change="vm.updateDialogFieldResponders(vm.modalData.name)"
```
Comment (himdel) [inline src/dialog-editor/components/modal-field/field.html]: Should we have a default implementation of `vm.updateDialogFieldResponders`? 

Unconditionally calling an optional function sounds potentially bad...

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 5f3f3f5eca6325b14a14871edb1fa7d0eaad66a1
Date:   Tue Sep 25 12:11:09 2018 +0000

    Merge pull request #347 from romanblanco/bz1614436
    
    Update dynamic fields responders after change
    
    (cherry picked from commit f9c976b39faaa06517c5ab481d238fbe28ecacc9)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1632769
```

**Comment by himdel** [comment]
Released in 1.0.36

**Pull request merged**